chains/evm/cli/admin/withdraw.go

```diff
@@ -14,99 +26,122 @@ var withdrawCmd = &cobra.Command{
+	// fetch global flag values
+	url, gasLimit, gasPrice, senderKeyPair, err := flags.GlobalFlagValues(cmd)
+	if err != nil {
+		return fmt.Errorf("could not get global flags: %v", err)
 	}
-	handlerAddress := common.HexToAddress(handler)
 
-	token := cctx.String("token")
-	if !common.IsHexAddress(token) {
-		return fmt.Errorf("invalid token address %s", token)
-	}
-	tokenAddress := common.HexToAddress(token)
+	fmt.Printf("Withdrawing %s token from handler: %s", Amount, Handler)
 
-	recipient := cctx.String("recipient")
-	if !common.IsHexAddress(recipient) {
-		return fmt.Errorf("invalid recipient address %s", recipient)
+	ethClient, err := evmclient.NewEVMClientFromParams(url, senderKeyPair.PrivateKey())
+	if err != nil {
+		log.Error().Err(fmt.Errorf("eth client initialization error: %v", err))
+		return err
 	}
-	recipientAddress := common.HexToAddress(recipient)
+	gasPricer.SetClient(ethClient)
+	gasPricer.SetOpts(&evmgaspricer.GasPricerOpts{UpperLimitFeePerGas: gasPrice})
 
-	amount := cctx.String("amount")
-	id := cctx.String("id")
+	// @dev withdrawal data should include:
+	// tokenAddr => common.Address
+	// recipientAddr => common.Address
+	// amountOrTokenID => uint256 (*big.Int)
+	withdrawalData := bytes.Buffer{}
+	withdrawalData.Write(common.LeftPadBytes(tokenAddr.Bytes(), 32))
+	withdrawalData.Write(common.LeftPadBytes(recipientAddr.Bytes(), 32))
+	withdrawalData.Write(common.LeftPadBytes(realAmount.Bytes(), 32))
 
-	if id != "" && amount != "" {
-		return errors.New("Only id or amount should be set.")
-	}
-	if id == "" && amount == "" {
-		return errors.New("id or amount flag should be set")
-	}
-	ethClient, err := client.NewClient(url, false, sender, big.NewInt(0).SetUint64(gasLimit), big.NewInt(0).SetUint64(gasPrice), big.NewFloat(1))
+	input, err := calls.PrepareWithdrawInput(handlerAddr, withdrawalData.Bytes())
 	if err != nil {
+		log.Error().Err(fmt.Errorf("admin withdrawal input error: %v", err))
 		return err
 	}
-	idOrAmountToWithdraw := new(big.Int)
-	if amount != "" {
-		decimals := big.NewInt(0).SetUint64(cctx.Uint64("decimals"))
-		idOrAmountToWithdraw, err = utils.UserAmountToWei(amount, decimals)
-		if err != nil {
-			return err
-		}
-	} else {
-		idOrAmountToWithdraw.SetString(id, 10)
+	blockNum, err := ethClient.BlockNumber(context.Background())
+	if err != nil {
+		log.Error().Err(fmt.Errorf("block fetch error: %v", err))
+		return err
 	}
 
-	err = utils.AdminWithdraw(ethClient, bridgeAddress, handlerAddress, tokenAddress, recipientAddress, idOrAmountToWithdraw)
+	log.Debug().Msgf("blockNum: %v", blockNum)
+
+	txHash, err := calls.Transact(ethClient, txFabric, gasPricer, &bridgeAddr, input, gasLimit, big.NewInt(0))
 	if err != nil {
-		return err
+		log.Error().Err(fmt.Errorf("admin withdrawal error: %v", err))
 	}
 
-	log.Info().Msgf("Withdrawn %s to %s", idOrAmountToWithdraw.String(), recipient)
+	log.Debug().Msgf("admin withdrawal hash: %s", txHash.Hex())
```

Comment:
Probably, just add hash to info log.
Users still need that information so no need for it to be debug.